scripts/windows/setup_project.bat

```diff
@@ -0,0 +1,252 @@
+   set deps_install_dir=%2
+   set /a set_deps_install=1
+   shift
+   shift
+   goto :interpret_arg
+   )
+   if /I "%1" EQU "-g" (
+   set generator=%2
+   set /a set_generator=1
+   shift
+   shift
+   goto :interpret_arg
+   )
+   if /I "%1" EQU "-c" (
+   set config_type=%2
+   set /a set_config=1
+   shift
+   shift
+   goto :interpret_arg
+   )
+   shift
+   goto :interpret_arg
+)
+
+if %display_help%==1 (
+   call :print_help
+   EXIT /B 0
+   )
+
+::check if the configuration is correct
+set /a opt=0
+if "%config_type%"=="Release" (
+    set /a opt=1
+)
+if "%config_type%"=="Debug" (
+    set /a opt=1
+)
+if %set_config%==0 ( 
+    set /a opt=1
+    set config_type=Release
+    )
+if %opt%==0 (
+    echo Please enter a correct configuration (Release or Debug^)
+    EXIT /B %ERRORLEVEL%
+)
+echo Setup will continue with the configuration: %config_type%
+
+
+::check if the generator is correct
+set /a opt=0
+if %generator%=="Visual Studio 15 2017 Win64" (
+    set /a opt=1
+)
+if %generator%=="Visual Studio 14 2015 Win64" (
+    set /a opt=1
+)
+if %generator%=="Visual Studio 15 2017" (
+    set /a opt=1
+)
+if %generator%=="Visual Studio 14 2015" (
+    set /a opt=1
+)
+if %set_generator%==0 (
+   set /a opt=1
+   set generator="Visual Studio 15 2017 Win64"
+   )
+if %opt%==0 (
+    echo Please enter a correct configuration ("Visual Studio 15 2017 Win64", "Visual Studio 15 2017", "Visual Studio 14 2015 Win64" or "Visual Studio 14 2015"^)
+    EXIT /B %ERRORLEVEL%
+)
+echo Setup will continue with the generator: %generator%
+
+::set the diretories
+if %set_build%==0 (
+   set build_dire=%CD%\build
+   )
+echo The sdk will be built in: %build_dire%
+
+if %set_deps%==0 (
+   set deps_dir=%CD%\deps
+   )
+echo The deps will be downloaded in: %deps_dir%
+
+if %set_deps_install%==0 (
+   set deps_install_dir=%deps_dir%\installed
+   )
+echo The deps will be installed in:  %deps_install_dir%
+
+::ask for permission to continue the setup
+if %answer_yes%==0 (
+   call :yes_or_exit "Do you want to continue?"
+   )
+
+set /P openSSLPath="Please enter OpenSSL root absolute path: "
+echo %openSSLPath%
+
+set /P openCVPath="Please enter opencv 3.4.1 root absolute path: "
+echo %openCVPath%
+
+::create the missing folders
+if not exist %build_dire% md %build_dire%
+if not exist %deps_dir% md %deps_dir%
+if not exist %deps_install_dir% md %deps_install_dir%
+
+::call functions that install the dependencies
+CALL :install_glog %config_type% %generator%
+CALL :install_protobuf %config_type% %generator%
+CALL :install_websockets %config_type% %generator%
+
+::build the project with the selected options
+set CMAKE_OPTIONS=-DWITH_PYTHON=on -DWITH_OPENCV=on
+pushd %build_dire%
+cmake -G %generator% -DWITH_PYTHON=on -DWITH_OPENCV=on -DOpenCV_DIR="%openCVPath%\build\x64\vc15\lib" -DCMAKE_PREFIX_PATH="%deps_install_dir%\glog;%deps_install_dir%\protobuf;%deps_install_dir%\libwebsockets" -DOPENSSL_INCLUDE_DIRS="%openSSLPath%\include" %source_dir%
```

Comment:
..\vc15\lib should not be hardcoded as it depends on the generator used. For 'Visual Studio 14 2015', path needs to be ..\vc14\lib\

Reply:
solved with commit: 91931cc